python-package/packager/pep517.py

```diff
@@ -0,0 +1,191 @@
+"""
+Custom build backend for XGBoost Python package.
+Builds source distribution and binary wheels, following PEP 517 / PEP 660.
+Re-uses components of Hatchling (https://github.com/pypa/hatch/tree/master/backend) for the sake
+of brevity.
+"""
+import dataclasses
+import logging
+import os
+import pathlib
+import sysconfig
+import tempfile
+from contextlib import contextmanager
+from typing import Any, Dict, Iterator, List, Optional, Union
+
+import hatchling.build
+
+from .build_config import BuildConfiguration
+from .nativelib import locate_local_libxgboost, locate_or_build_libxgboost
+from .sdist import copy_cpp_src_tree
+from .util import copy_with_logging, copytree_with_logging
+
+
+@contextmanager
+def cd(path: Union[str, pathlib.Path]) -> Iterator[str]:  # pylint: disable=C0103
+    """Temporarily change working directory"""
+    if isinstance(path, pathlib.Path):
+        path = str(path)
+    path = os.path.realpath(path)
+    cwd = os.getcwd()
+    os.chdir(path)
+    try:
+        yield path
+    finally:
+        os.chdir(cwd)
+
+
+def get_tag() -> str:
+    """Get appropate wheel tag, according to system"""
+    tag_platform = sysconfig.get_platform().replace("-", "_").replace(".", "_")
+    return f"py3-none-{tag_platform}"
+
+
+TOPLEVEL_DIR = pathlib.Path(__file__).parent.parent.absolute().resolve()
+logging.basicConfig(level=logging.INFO)
+
+
+def get_requires_for_build_wheel(
+    config_settings: Optional[Dict[str, Any]] = None
+) -> List[str]:
+    """A PEP 517 method. Delegate to Hatchling"""
+    return hatchling.build.get_requires_for_build_wheel(config_settings)
+
+
+def get_requires_for_build_sdist(
+    config_settings: Optional[Dict[str, Any]] = None
+) -> List[str]:
+    """A PEP 517 method. Delegate to Hatchling"""
+    return hatchling.build.get_requires_for_build_sdist(config_settings)
+
+
+def get_requires_for_build_editable(
+    config_settings: Optional[Dict[str, Any]] = None
+) -> List[str]:
+    """A PEP 517 method. Delegate to Hatchling"""
+    return hatchling.build.get_requires_for_build_editable(config_settings)
+
+
+def write_hatch_config(dest_dir: pathlib.Path, *, logger: logging.Logger) -> None:
```

Comment:
Rather than writing this file as a temporary, can't you just commit the `hatch_build.py` file to the repo with the `get_tag` function in that file to be used by the `initialize` function. Then the file would always be present and it would be copied into the temporary dir in `build_wheel` too, right?

Reply:
Good point, let me do that.